x-pack/filebeat/input/httpjson/config_auth.go

```diff
@@ -109,14 +111,30 @@ func (o *oAuth2Config) client(ctx context.Context, client *http.Client) (*http.C
 
 	switch o.getProvider() {
 	case oAuth2ProviderAzure, oAuth2ProviderDefault:
-		creds := clientcredentials.Config{
-			ClientID:       o.ClientID,
-			ClientSecret:   o.ClientSecret,
-			TokenURL:       o.getTokenURL(),
-			Scopes:         o.Scopes,
-			EndpointParams: o.getEndpointParams(),
+		if o.User != "" || o.Password != "" {
+			conf := &oauth2.Config{
+				ClientID:     o.ClientID,
+				ClientSecret: o.ClientSecret,
+				Endpoint: oauth2.Endpoint{
+					TokenURL:  o.TokenURL,
+					AuthStyle: 1,
+				},
+			}
+			token, err := conf.PasswordCredentialsToken(ctx, o.User, o.Password)
+			if err != nil {
+				return nil, fmt.Errorf("oauth2 client: error loading credentials: %w", err)
```

Comment:
nit: It would be valuable to give more context here (`error loading credentials using user and password`).

Reply:
Got it.
I will update it.